inc/includes.php

```diff
@@ -181,4 +181,17 @@
 // Manage entity change
 if (isset($_REQUEST["force_entity"]) && ($_SESSION["glpiactive_entity"] ?? -1) != $_REQUEST["force_entity"]) {
     Session::changeActiveEntities($_REQUEST["force_entity"], true);
+} else {
+    $glpiactiveentities = $_SESSION['glpiactiveentities'] ?? [];
+    if (count($glpiactiveentities) > 1) {
+        $entities = getSonsOf("glpi_entities", $_SESSION["glpiactive_entity"]);
+        sort($entities);
+        sort($glpiactiveentities);
+        if ($entities != $glpiactiveentities) {
```

Comment:
The following check should be really faster:
```suggestion
        if (
            count($entities) !== count($glpiactiveentities)
            && array_diff($entities, $glpiactiveentities) === []
            && array_diff($glpiactiveentities, $entities) === []
        ) {
```

Reply:
I had also considered using count(), but it doesn't account for the case where the admin swaps two entities.

Reply:
Indeed. The condition was not correct anyway. Using `||` operators would permit to not have to check for differences if the count in different, and that would optimize the check.
```php
        if (
            count($entities) !== count($glpiactiveentities)
            || array_diff($entities, $glpiactiveentities) !== []
            || array_diff($glpiactiveentities, $entities) !== []
        ) {
```